### DISCUSSION TITLE
Problems with install script

### DISCUSSION THREAD

**Discussion by rdfry** [OpenHantek/OpenHantek6022#187] [Q&A] [answered]
I am trying to install openhantek in Sparkylinux. I used the install script and I get a Cpack error. It says cannot install 60-hantek.rules in /lib/udeve/rules.d/ directory does not exist. However if I open rules.d it is there.

**Comment by Ho-Ro** [comment]
> I used the install script and I get a Cpack error. 

Which script? The normal way to install for a debian based distro is to use the *.deb, in best case [built on your system](https://github.com/OpenHantek/OpenHantek6022#building-openhantek6022-from-source), but you can also use the [deb package from a release](https://github.com/OpenHantek/OpenHantek6022#install-prebuilt-binary-packages). 

> It says cannot install 60-hantek.rules in /lib/udeve/rules.d/ directory does not exist.

Please show the complete script call and the actual error message instead of just paraphrasing it.

**Reply by rdfry** [reply]
Using the Linux install script for openhantek-master.
Output is : 100% Built target OpenHantek
Run CPack packaging tool.... 
CPack: Create package using STGZ
CPack: Install projects
CPack: - Run preinstall target for:  OpenHantekProject
CPack: - Install project: OpenHantekProject
Cmake Error at /home/danf/build/openhantek-master/build/cmake_install.cmake:49 (file) : file Install cannot copy file "/home/danf/build/openhantek-master/firmware/60-hantek.rules" to /lib/udev/rules.d/60-hantek.rules"

CPack Error: Error when generating package: openhantekproject
make: *** [Makefile:108: package] Error 1

**Reply by Ho-Ro** [reply]
**Are you sure that you are talking about OpenHantek6022?**

Please tell exactly what you are doing, which command do you type in???

> CPack: Create package using STGZ
CPack: Install projects
CPack: - Run preinstall target for: OpenHantekProject
CPack: - Install project: OpenHantekProject

From your output I see that you are doing something different - STGZ vs. TGZ; OpenHantekProject vs. OpenHantek, please compare the output of my build:

````
git clone https://github.com/OpenHantek/OpenHantek6022.git
Cloning into 'OpenHantek6022'...
remote: Enumerating objects: 9540, done.
remote: Counting objects: 100% (689/689), done.
remote: Compressing objects: 100% (280/280), done.
remote: Total 9540 (delta 456), reused 582 (delta 375), pack-reused 8851
Receiving objects: 100% (9540/9540), 66.47 MiB | 6.17 MiB/s, done.
Resolving deltas: 100% (7478/7478), done.
````
````
cd OpenHantek6022
sh LinuxBuild
-- The C compiler identification is GNU 8.3.0
-- The CXX compiler identification is GNU 8.3.0
-- Check for working C compiler: /bin/cc
-- Check for working C compiler: /bin/cc -- works
...
-- Build files have been written to: ...
````
...
````
[100%] Built target OpenHantek
Run CPack packaging tool...
CPack: Create package using TGZ
CPack: Install projects
CPack: - Run preinstall target for: OpenHantek
CPack: - Install project: OpenHantek
CPack: Create package
CPack: - package: packages/openhantek_20210526-ab9a649-1_x86_64.tar.gz generated.
CPack: Create package using DEB
CPack: Install projects
CPack: - Run preinstall target for: OpenHantek
CPack: - Install project: OpenHantek
CPack: Create package
CPack: - package: packages/openhantek_20210526-ab9a649_amd64.deb generated.
CPack: Create package using RPM
CPack: Install projects
CPack: - Run preinstall target for: OpenHantek
CPack: - Install project: OpenHantek
CPack: Create package
-- alien found, we may be on a Debian based distro.
-- CPackRPM:Debug: Using CPACK_RPM_ROOTDIR=/home/horo/projects/Measure/Hantek6022/TEST/OpenHantek6022/build/_CPack_Packages/Linux/RPM
CPackRPM: Will use GENERATED spec file: /home/horo/projects/Measure/Hantek6022/TEST/OpenHantek6022/build/_CPack_Packages/Linux/RPM/SPECS/openhantek.spec
CPack: - package: packages/openhantek-20210526_ab9a649-1.x86_64.rpm generated.

Packages are available in build/packages:
build/packages/openhantek-20210526_ab9a649-1.x86_64.rpm
build/packages/openhantek_20210526-ab9a649-1_x86_64.tar.gz
build/packages/openhantek_20210526-ab9a649_amd64.deb
````
> Cmake Error at /home/danf/build/openhantek-master/build/cmake_install.cmake:49 (file) : file Install cannot copy file "/home/danf/build/openhantek-master/firmware/60-hantek.rules" to /lib/udev/rules.d/60-hantek.rules"

EDIT:
Even if you are using a different OH project, your error message indicates missing write permissions - was your call `fakeroot make package` (preferred) or `sudo make package`?

**Reply by rdfry** [reply]
I am using the Linuxbuild script listed in Openhantek-master code. Copy and paste in terminal in the Openhantek-master folder without root.

**Comment by Ho-Ro** [comment, marked as answer]
> I am using the Linuxbuild script listed in Openhantek-master code. Copy and paste in terminal in the Openhantek-master folder without root.

WHY??? The file is still there in your local git repo - you made a local clone from your fork - didn't you? 

I hate to repeat myself - **you are not talking about current OpenHantek6022!**
- OpenHantek6022 does not use **STGZ**, only **TGZ** packaging:
`git grep TGZ`
`cmake/CPackInfos.cmake:    set(CPACK_GENERATOR TGZ)`
STGZ packaging was [removed on 1-Mar-2019](https://github.com/OpenHantek/OpenHantek6022/commit/e4e849f2943d36d86655fcb5d6bfb73858a90af4#diff-a07c1f0815917babb582734c2c93b72a8228046a07a8eec0bbcb20961b8e3088L71) 
- OpenHantek6022 does not have a branch named `master` - the trunk is called `main`.
- If you download [the zip from your fork](https://github.com/rdfry/OpenHantek6022/archive/refs/heads/main.zip) instead of cloning (WHY???) it expands to the directory `OpenHantek6022-main` and not `openhantek-master`. Obviously you're talking about [openhantek](https://github.com/OpenHantek/openhantek), but unfortunately this project is ~dead~ [currently unmaintained](https://github.com/OpenHantek/openhantek/issues/277).

Please do me - and especially you - a favour and start with a clean install, either from my repo or from your fork:
````
cd
mkdir OpenHantekTest
cd OpenHantekTest
git clone https://github.com/rdfry/OpenHantek6022.git
cd OpenHantek6022
bash LinuxBuild
````
If you use a different shell, e.g. `/bin/dash` then you should change [line 5](https://github.com/rdfry/OpenHantek6022/blob/3af5b2cc5457908a6d753771bdeaaaac207fcbda/LinuxBuild#L5)
from`if [ $(id -u) == 0 ]; then` to `if [ $(id -u) = 0 ]; then`.
You should end up with three files in `build/packages`:
````
-rw-r--r-- 1 horo horo 5111054 Mai 27 09:09 openhantek-20210527_ab9a649-1.x86_64.rpm
-rw-r--r-- 1 horo horo 5105123 Mai 27 09:09 openhantek_20210527-ab9a649-1_x86_64.tar.gz
-rw-r--r-- 1 horo horo 5106898 Mai 27 09:09 openhantek_20210527-ab9a649_amd64.deb
````
**I will not provide any further support until you have performed this test exactly as described!**

**Comment by rdfry** [comment]
OK I appreciate your helping me and will give that a go. 
This is what I have been looking at in the open hantek 6022 repo and ran linuxsetup_asroot and Linux build scripts
Main
Folder githubFolder cmakeFolder docsFolder openhantekFolder utils
            editorconfig             . gitmore            . gitmodules            . travis_inactive             changelog             cmakelists. txt             copying             doxyfile             license             linuxbuild             linuxsetup_asroot             readme             readme. md             appveyor_inactive
    On Thursday, May 27, 2021, 01:11:09 PM CDT, Martin ***@***.***> wrote:  
 
 



I am using the Linuxbuild script listed in Openhantek-master code. Copy and paste in terminal in the Openhantek-master folder without root.


WHY??? The file is still there in your local git repo - you made a local clone from your fork - didn't you?

I hate to repeat myself - you are not talking about current OpenHantek6022!
   
   - OpenHantek6022 does not use STGZ, only TGZ packaging:   
git grep TGZ   
cmake/CPackInfos.cmake: set(CPACK_GENERATOR TGZ)   
STGZ packaging was removed on 1-Mar-2019
   - OpenHantek6022 does not have a branch named master - the trunk is called main.
   - If you download the zip from your fork instead of cloning (WHY???) it expands to the directory OpenHantek6022-main and not openhantek-master. Obviously you're talking about openhantek, but unfortunately this project is     currently unmaintained.

Please do me - and especially you - a favour and start with a clean install, either from my repo or from your fork:
cd
mkdir OpenHantekTest
cd OpenHantekTest
git clone https://github.com/rdfry/OpenHantek6022.git
cd OpenHantek6022
bash LinuxBuild

If you use a different shell, e.g. /bin/dash then you should change line 5
fromif [ $(id -u) == 0 ]; then to if [ $(id -u) = 0 ]; then.
You should end up with three files in build/packages:
-rw-r--r-- 1 horo horo 5111054 Mai 27 09:09 openhantek-20210527_ab9a649-1.x86_64.rpm
-rw-r--r-- 1 horo horo 5105123 Mai 27 09:09 openhantek_20210527-ab9a649-1_x86_64.tar.gz
-rw-r--r-- 1 horo horo 5106898 Mai 27 09:09 openhantek_20210527-ab9a649_amd64.deb

I will not provide any further support until you have performed this test exactly as described!

**Comment by rdfry** [comment]
Followed your instructions and have the three build packages. 
We had a miscommunication previously, I never used a forked version it was OpenHantek 6022 from your Repo. For some reason LinuxBuild was trying to use Stgz that is what is confusing
    On Thursday, May 27, 2021, 01:11:09 PM CDT, Martin ***@***.***> wrote:  
 
 



I am using the Linuxbuild script listed in Openhantek-master code. Copy and paste in terminal in the Openhantek-master folder without root.


WHY??? The file is still there in your local git repo - you made a local clone from your fork - didn't you?

I hate to repeat myself - you are not talking about current OpenHantek6022!
   
   - OpenHantek6022 does not use STGZ, only TGZ packaging:   
git grep TGZ   
cmake/CPackInfos.cmake: set(CPACK_GENERATOR TGZ)   
STGZ packaging was removed on 1-Mar-2019
   - OpenHantek6022 does not have a branch named master - the trunk is called main.
   - If you download the zip from your fork instead of cloning (WHY???) it expands to the directory OpenHantek6022-main and not openhantek-master. Obviously you're talking about openhantek, but unfortunately this project is     currently unmaintained.

Please do me - and especially you - a favour and start with a clean install, either from my repo or from your fork:
cd
mkdir OpenHantekTest
cd OpenHantekTest
git clone https://github.com/rdfry/OpenHantek6022.git
cd OpenHantek6022
bash LinuxBuild

If you use a different shell, e.g. /bin/dash then you should change line 5
fromif [ $(id -u) == 0 ]; then to if [ $(id -u) = 0 ]; then.
You should end up with three files in build/packages:
-rw-r--r-- 1 horo horo 5111054 Mai 27 09:09 openhantek-20210527_ab9a649-1.x86_64.rpm
-rw-r--r-- 1 horo horo 5105123 Mai 27 09:09 openhantek_20210527-ab9a649-1_x86_64.tar.gz
-rw-r--r-- 1 horo horo 5106898 Mai 27 09:09 openhantek_20210527-ab9a649_amd64.deb

I will not provide any further support until you have performed this test exactly as described!

**Reply by Ho-Ro** [reply]
> Followed your instructions and have the three build packages. 
We had a miscommunication previously, I never used a forked version it was OpenHantek 6022 from your Repo. For some reason LinuxBuild was trying to use Stgz that is what is confusing

![image](https://user-images.githubusercontent.com/12542359/120153287-d2ec2280-c1ee-11eb-8225-ba4bb34e6494.png)

This shows the importance (for everyone who asks for support) to clearly state what was done and especially what is **your intention**, e.g.:
- Intention: I want OpenHantek6022 installed on my SparkyLinux system.
Solution: As SparkyLinux is based on Debian just use the pre built `*.deb` release package.
- Intention: I want to modify some features of OH6022.
Solution: Clone this repo locally, do the change and use the build script.
- Intention: I'm sure I found an error in OH6022.
Solution: Describe exactly what you did to trigger the error - use the provided [issue template](https://github.com/OpenHantek/OpenHantek6022/issues/new?assignees=&labels=&template=bug_report.md&title=).
- Intention: I want to install the [old openhantek](https://github.com/OpenHantek/openhantek) because it has support for other scopes not supported in OH6022.
Solution: Sorry, no support for other HW here, you have to investigate how the project works (this is what I did to fully support Hantek6022 more than 2 years ago).

Otherwise we end up in an [XY problem](https://en.wikipedia.org/wiki/XY_problem).

**Comment by rdfry** [comment]
Thanks I appreciate you. And for your work on this application
    On Monday, May 31, 2021, 02:25:25 AM CDT, Martin ***@***.***> wrote:  
 
 



Followed your instructions and have the three build packages.
We had a miscommunication previously, I never used a forked version it was OpenHantek 6022 from your Repo. For some reason LinuxBuild was trying to use Stgz that is what is confusing




This shows the importance (for everyone who asks for support) to clearly state what was done and especially what is your intention, e.g.:
   
   - Intention: I want OpenHantek6022 installed on my SparkyLinux system.   
Solution: As SparkyLinux is based on Debian just use the pre built *.deb release package.
   - Intention: I want to modify some features of OH6022.   
Solution: Clone this repo locally, do the change and use the build script.
   - Intention: I'm sure I found an error in OH6022.   
Solution: Describe exactly what you did to trigger the error - use the provided issue template.
   - Intention: I want to install the old openhantek because it has support for other scopes not supported in OH6022.   
Solution: Sorry, no support for other HW here, you have to investigate how the project works (this is what I did to fully support Hantek6022 more than 2 years ago).

Otherwise we end up in an XY problem.

**Comment by rdfry** [comment]
Purchased your Hscope app so Thankyou for that too
    On Monday, May 31, 2021, 02:25:25 AM CDT, Martin ***@***.***> wrote:  
 
 



Followed your instructions and have the three build packages.
We had a miscommunication previously, I never used a forked version it was OpenHantek 6022 from your Repo. For some reason LinuxBuild was trying to use Stgz that is what is confusing




This shows the importance (for everyone who asks for support) to clearly state what was done and especially what is your intention, e.g.:
   
   - Intention: I want OpenHantek6022 installed on my SparkyLinux system.   
Solution: As SparkyLinux is based on Debian just use the pre built *.deb release package.
   - Intention: I want to modify some features of OH6022.   
Solution: Clone this repo locally, do the change and use the build script.
   - Intention: I'm sure I found an error in OH6022.   
Solution: Describe exactly what you did to trigger the error - use the provided issue template.
   - Intention: I want to install the old openhantek because it has support for other scopes not supported in OH6022.   
Solution: Sorry, no support for other HW here, you have to investigate how the project works (this is what I did to fully support Hantek6022 more than 2 years ago).

Otherwise we end up in an XY problem.
